src/sage/features/__init__.py

```diff
@@ -791,26 +792,37 @@ class StaticFile(FileFeature):
     EXAMPLES::
 
         sage: from sage.features import StaticFile
-        sage: StaticFile(name="no_such_file", filename="KaT1aihu", search_path=("/",), spkg="some_spkg", url="http://rand.om").require()  # optional - sage_spkg
+        sage: StaticFile(name="no_such_file", filename="KaT1aihu",              # optional - sage_spkg
+        ....:            search_path="/", spkg="some_spkg",
+        ....:            url="http://rand.om").require()
         Traceback (most recent call last):
         ...
         FeatureNotPresentError: no_such_file is not available.
         'KaT1aihu' not found in any of ['/']...
         To install no_such_file...you can try to run...sage -i some_spkg...
         Further installation instructions might be available at http://rand.om.
     """
-    def __init__(self, name, filename, search_path=None, **kwds):
+    def __init__(self, name, filename, *, search_path=None, **kwds):
         r"""
         TESTS::
 
             sage: from sage.features import StaticFile
-            sage: StaticFile(name="null", filename="null", search_path=("/dev",))
+            sage: StaticFile(name="null", filename="null", search_path="/dev")
             Feature('null')
+            sage: sh = StaticFile(name="shell", filename="sh",
+            ....:                 search_path="/dev:/bin:/usr")
+            sage: sh
+            Feature('shell')
+            sage: sh.absolute_filename()
+            '/bin/sh'
+
         """
         Feature.__init__(self, name, **kwds)
         self.filename = filename
         if search_path is None:
             self.search_path = [SAGE_SHARE]
+        elif isinstance(search_path, str):
+            self.search_path = search_path.split(":")
```

Comment:
is this used somewhere?
splitting on colon is a potential porting hazard when native Windows paths get involved

Reply:
No, I don't think I'm using it right now, but might use it later so I prefer to state the intention now... Using `:` as separators is standard, and we need this for `SAGE_DATA_PATH`. We could use `os.pathsep`, but I think ":" is already used explicitly in a some places (e.g. `Executable.absolute_filename()`).

Internally we can use lists, but this is meant to communicate through environment vaiables so there is need to serialize.

How does WSL solve this?

Reply:
I don't see why this would have to be handled on the level of the input args of the `Feature`. The input args have Python types.